packages/mobile/src/hooks/useFetchAllFavoritedTracks.ts

```diff
@@ -18,36 +18,32 @@ const { getLocalSaves } = savedPageSelectors
 
 const { getUserId } = accountSelectors
 
+type FavoritedTracksResponse = {
+  data: {
+    favorite_item_id: string
+    favorite_type: string
+    created_at: string
+  }[]
+}
+
 export const fetchAllFavoritedTracks = async (currentUserId: number) => {
-  let tracksAndTimestamps: { trackId: ID; favoriteCreatedAt: string }[] = []
-  let loadMore = true
-  let offset = 0
-  // TODO: store results in state to avoid duplicate fetching
-  while (loadMore) {
```

Comment:
I assume we're okay with a hard cap of 10k downloaded favorites? We could keep the while loop but it won't come up much

Reply:
yeah i think lets just cap it!